apps/shipments/models.py

```diff
@@ -341,3 +343,98 @@ def update_vault_hash(self, vault_hash, rate_limit=False):
     VALID_UNTIL = 24
     FUNDING_TYPE = FundingType.SHIP.value
     SHIPMENT_AMOUNT = 1
+
+
+class TrackingData(models.Model):
+    id = models.CharField(primary_key=True, default=random_id, max_length=36)
+    created_at = models.DateTimeField(auto_now_add=True)
+    device_id = models.CharField(blank=True, null=True, max_length=36)
+    shipment = models.ForeignKey(Shipment, on_delete=models.PROTECT, null=True)
+    latitude = models.FloatField(default=None)
+    longitude = models.FloatField(default=None)
+    altitude = models.FloatField(default=None)
+    source = models.CharField(max_length=10)
+    uncertainty = models.IntegerField(blank=True, null=True)
+    speed = models.IntegerField(blank=True, null=True)
+    timestamp = models.CharField(max_length=50)
+    geometry = GeometryField(null=True)
```

Comment:
Why is this nullable? I think it should be automatically populated when a tracking data object is created. The only time it could be null is if latitude/longitude is null, which would mean the tracking data is bad and the whole POST should be rejected.